app/models/concerns/orchestration.rb

```diff
@@ -119,6 +126,16 @@ def process(queue_name)
     # if we have no failures - we are done
     return true if q.failed.empty? and q.pending.empty? and q.conflict.empty? and orchestration_errors?
 
+    fail_queue(q)
+
+    rollback
+  end
+
+  def skip_orchestration
+    Rails.env == "test"
```

Comment:
sorry if it's out of the scope, but could you please add following?

```ruby
def skip_orchestration!
  @skip_orchestration = true
end

def skip_orchestration? # replacing skip_orchestration
  @skip_orchestration ||= Rails.env == "test"
end
```

there's already `importing_facts` flag but I think it's not disabling all orchestration and this would be very useful